tests/integration/adapter-identifier-tests.js

```diff
@@ -0,0 +1,308 @@
+          id: '12d416e92eff0e8543ede886919c02c2',
+          identifiers: {
+            headers: { 'content-type': 'application/json;charset=utf-8' },
+            method: 'POST',
+            body: '{}',
+            url: 'http://a:b@localhost:4000?query=param'
+          }
+        },
+        xhr: {
+          id: '12d416e92eff0e8543ede886919c02c2',
+          identifiers: {
+            headers: { 'content-type': 'application/json;charset=utf-8' },
+            method: 'POST',
+            body: '{}',
+            url: 'http://a:b@localhost:4000?query=param'
+          }
+        }
+      });
+
+      testConfiguration('url.port', false, {
+        fetch: {
+          id: '147ca2615501f226981619c32341699b',
+          identifiers: {
+            headers: { 'content-type': 'application/json;charset=utf-8' },
+            method: 'POST',
+            body: '{}',
+            url: 'http://a:b@localhost/pathname?query=param'
+          }
+        },
+        xhr: {
+          id: '147ca2615501f226981619c32341699b',
+          identifiers: {
+            headers: { 'content-type': 'application/json;charset=utf-8' },
+            method: 'POST',
+            body: '{}',
+            url: 'http://a:b@localhost/pathname?query=param'
+          }
+        }
+      });
+
+      testConfiguration('url.query', false, {
+        fetch: {
+          id: '0678c78c38e1398f5d6e56d92b4a7647',
+          identifiers: {
+            headers: { 'content-type': 'application/json;charset=utf-8' },
+            method: 'POST',
+            body: '{}',
+            url: 'http://a:b@localhost:4000/pathname'
+          }
+        },
+        xhr: {
+          id: '0678c78c38e1398f5d6e56d92b4a7647',
+          identifiers: {
+            headers: { 'content-type': 'application/json;charset=utf-8' },
+            method: 'POST',
+            body: '{}',
+            url: 'http://a:b@localhost:4000/pathname'
+          }
+        }
+      });
+
+      testConfiguration('url.hash', true, {
+        fetch: {
+          id: 'a21ab01cc69a91e5312239023ea719b8',
+          identifiers: {
+            headers: { 'content-type': 'application/json;charset=utf-8' },
+            method: 'POST',
+            body: '{}',
+            url: 'http://a:b@localhost:4000/pathname?query=param#abc'
+          }
+        },
+        xhr: {
+          id: 'a21ab01cc69a91e5312239023ea719b8',
+          identifiers: {
+            headers: { 'content-type': 'application/json;charset=utf-8' },
+            method: 'POST',
+            body: '{}',
+            url: 'http://a:b@localhost:4000/pathname?query=param#abc'
+          }
+        }
+      });
+
+      testConfiguration('url.username', false, {
+        fetch: {
+          id: '3dbff3c3fbccd6f97d01be31fc7fdd59',
+          identifiers: {
+            headers: { 'content-type': 'application/json;charset=utf-8' },
+            method: 'POST',
+            body: '{}',
+            url: 'http://localhost:4000/pathname?query=param'
```

Comment:
Looks like we have a bug in Polly, url here dropped `password` from the matchBy.  Here we only turned off username, but password went with it.

Reply:
Good catch. We shoud create a new issue with this and follow up on this unless you have the time to fix it with this PR.

Reply:
Looks like the spec doesn't support passing credentials in the URL - results in a TypeError.  I think we should deprecate and eventually remove url.password and url.username options.